File: ruby/difference-of-squares/difference_of_squares.rb
# Class Squares
class Squares
  def initialize(number)
    @number = number
  end

  def sum_of_squares
    (1..@number).sum { |v| v**2 }
  end

  def square_of_sum
    (1..@number).sum**2
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

module BookKeeping
  VERSION = 4
end

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

import (
	"math"
)

// SumOfSquares calculate the sum of squares
func SumOfSquares(number int) int {
	sums := 0
	for i := 1; i <= number; i++ {
		sums += int(math.Pow(float64(i), 2))
	}
	return sums
}

// SquareOfSums calculate the square of sums
func SquareOfSums(number int) int {
	sums := 0
	for i := 1; i <= number; i++ {
		sums += i
	}
	return int(math.Pow(float64(sums), 2))
}

// Difference of SquareOfSums SumOfSquares
func Difference(number int) int {
	return SquareOfSums(number) - SumOfSquares(number)
}
